### DISCUSSION TITLE
Babel config is not used (preset-react)

### DISCUSSION THREAD

**Discussion by SimonChaumet** [vercel/next.js#33737] [Help]
**What version of Next.js are you using?**

9.5.2 and 10.0.7

**What version of Node.js are you using?**

12.19.0

**What browser are you using?**

Firefox, Chromium

**What operating system are you using?**

Arch linux

**How are you deploying your application?**

docker

**Describe the Bug**

When setting the babel config in .babelrc.js or .babelrc the settings for optimization are not taken into consideration.

In my case I want to keep react display names when env is preproduction.

**Expected Behavior**

Settings are taken into considerations and display names are not removed.

**To Reproduce**

- Create a next project with babel or with Nx
- add configuration to next/babel preset
- build.

.babelrc.js :
```javascript
module.exports = {
  'presets': [
    ['next/babel', {
      'preset-react': {
        development: true,
      },
    }]
  ],
  "plugins": [
    [
      "styled-components",
      {
        "pure": true,
        "ssr": true
      }
    ]
  ]
}
```

Logs indicates that the right .babelrc.js is used.

**Comment by aacoelho** [comment]
I had a similar problem - `babel.config.js` was detected by next but wasn't used. I followed this https://github.com/vercel/next.js/issues/19334#issuecomment-748569767 reply and it worked. Calling `next/babel` and the plugin with `require.resolve` solves my problem.

my `babel.config.js` :
```js
module.exports = {
  "presets": [
    [require.resolve('next/babel')]
  ],
  "plugins": [
    [require.resolve("@emotion/babel-plugin")]
  ]
}
```
